tests/test_training/test_opt.py

```diff
@@ -192,65 +187,100 @@ def cost_fn():
 def test_learning_four_mode_Interferometer():
     """Finding the optimal Interferometer to make a NOON state with N=2"""
     settings.SEED = 4
+    solution_U = np.array(
```

Comment:
Is this the numerical solution or an analytical solution?

Reply:
numerical